homeassistant/components/sensor/velo_antwerpen.py

```diff
@@ -0,0 +1,107 @@
+"""
+Shows the available amount of public city bikes for Velo Antwerpen.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.velo_antwerpen/
+"""
+
+import logging
+
+import requests
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME, ATTR_ATTRIBUTION
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Velo'
+DEFAULT_ICON = 'mdi:bike'
+DEFAULT_UNIT = 'bikes'
+
+CONF_STATION_ID = 'station_id'
+
+RESOURCE_URL = "https://www.velo-antwerpen.be/availability_map/getJsonObject"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_STATION_ID): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Velo sensor."""
+    name = config.get(CONF_NAME)
+    station = config.get(CONF_STATION_ID)
+
+    add_entities([VeloSensor(name, station)], True)
+
+
+class VeloSensor(Entity):
+    """Get the available amount of bikes and set the selected station as attributes"""
+
+    def __init__(self, name, station):
+        """Initialize the Velo sensor."""
+        self._name = name
+        self._station = station
+        self._state = None
+        self._station_data = self.make_request()
```

Comment:
I think this `make_request()` call is not needed because you pass `True` to `add_entities`.

Reply:
True, I should have called self.update here instead I think.
This fn was called to make sure I have the data from the API so the attributes could be set before update was called by Home Assistant on the sensor itself.

Reply:
When you pass the `True` parameter to `add_entities`, it will call `update()` for you. And it seems like `update()` will in turn call `make_request()`.

But now I see that `update()` does not store the result to `self._station_data` so the initial call here is actually in use after all :)